### ISSUE TITLE
util.SharedRandom returns diffrent results in hooks

### ISSUE THREAD

**Issue by devil123w** [Facepunch/garrysmod-issues#5943]
###Details
I tested this on the 64-bit branch of Garry's Mod.

The util.SharedRandom function should return the same value based on the input seed and additional seed. However, it produces random numbers with the same input seeds in hooks, which should not be the case according to the wiki example.

### Steps to Reproduce
To reproduce this issue, run the following code in a server-side file and trigger the PlayerDeath hook:

`print(util.SharedRandom("MyString", 1000, 9999, 10))`

**Comment by robotboy655** [comment]
No, a "random" function that produces the same output every time is illogical.

It will produce the same output for a given moment in time (in a predicted hook) on both client and server. That is it's purpose, for prediciton.

Outside of predicted hooks I not sure how it will behave.
